File: Binary Trees/Traversals/07-Level Order Traversal.java
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if(root == null){
            return new ArrayList<>();
        }

        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);

        List<List<Integer>> traversal = new ArrayList<>();
        
        while(!q.isEmpty()){
            List<Integer> level = new ArrayList<>();
            int n = q.size();

            for(int i = 0; i < n; i++){
                TreeNode temp = q.poll();

                level.add(temp.val);

                if(temp.left != null){
                    q.offer(temp.left);
                }
                if(temp.right != null){
                    q.offer(temp.right);
                }
            }

            traversal.add(level);
        }

        return traversal;
    }
}

File: Binary Trees/Traversals/07-Level Order Traversal.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(root == nullptr){
            return {};
        }

        vector<vector<int>> traversal;
        
        queue<TreeNode*> q;
        q.push(root);

        while(!q.empty()){
            vector<int> level;
            int n = q.size();

            for(int i = 0; i < n; i++){
                TreeNode* temp = q.front();
                q.pop();

                level.push_back(temp -> val);

                if(temp -> left){
                    q.push(temp -> left);
                }
                if(temp -> right){
                    q.push(temp -> right);
                }
            }

            traversal.push_back(level);
        }

        return traversal;
    }
};